### DISCUSSION TITLE
subjects as a hashtag

### DISCUSSION THREAD

**Discussion by GuilhermeDeLucca** [LiveHelperChat/livehelperchat#1896] [General]
I believe this is already built in, and sorry for such a dumb question! but can i use the subjects as a hashtag?

It would be to use them as custom_string

**Comment by remdex** [comment]
I"m not sure I understood. Hashtags are only for canned messages at the moment
https://doc.livehelperchat.com/docs/canned

Subjects are completely different story. If you want just with `#hastag` E.g add subject to the chat quick way. You can just have canned messages with subject attached to it. So if operator sends specific message subject get's added automatically.

![image](https://user-images.githubusercontent.com/1146085/215695896-d4011d2c-5585-4830-bc82-64fe6eafcb17.png)

**Comment by GuilhermeDeLucca** [comment]
I had thought of something like that for example

![Captura de Tela 2023-01-31 às 09 50 01](https://user-images.githubusercontent.com/9898902/215764790-37640810-b1c3-4006-8577-3e7b9faa1fde.png)


The #hastag I'm referring to would be used as an identifier for what a visitor has done! Off topic!


It would be more for reporting and flow control (used in the bot)

It could know that a visitor identified himself in the system for example via API after asking a question X, and later he passed the product ID that was also located in the system! or it could have another #tag even though it says it was not found

Basically I would know that whoever requested something did what was requested by the bot

![Captura de Tela 2023-01-31 às 09 50 11](https://user-images.githubusercontent.com/9898902/215764826-df978da0-8250-43ca-9819-82b01972341b.png)

**Reply by remdex** [reply]
You can add subjects via bot also. https://doc.livehelperchat.com/docs/bot/bot-update-current-chat#set-subject

**Reply by GuilhermeDeLucca** [reply]
Yes I can add subjects with the bot! I know that I have internal affairs and general purpose affairs that can be used by the bot.

What I was asked would be if using the internal would use the class badge bg-warning mx-1 instead of badge bg-info mx-1 (I know I haven't tested it yet)

What I was wondering was if there was something really designed for this in the project even for use in reports (eg filtering a report by the "tag" product located)

I will create subjects and put # in front of the subject to symbolize that it is not a subject but a process note

**Reply by remdex** [reply]
Internal subject as you told was used just for bot as you ques. But they don't have specific class at the moment. You can E.g

Create a pull request with that improvement should not be hard :)
